grails-app/domain/au/org/emii/aatams/ReceiverDeployment.groovy

```diff
@@ -140,46 +133,99 @@ class ReceiverDeployment
+        // All non-null deployment intervals for receiver, other than 'this' deployment's.
+        def conflictingDeployments = deployment.receiver?.deployments.findAll {
+            !it.same(deployment) && it.undeployableInterval
+        }.findAll {
+            def checkInterval = it.undeployableInterval
+            checkInterval.contains([
+                deployment.initialisationDateTime,
+                deployment.deploymentDateTime
+            ]) ||
+
+            // deployment.deployedInterval?.overlaps(checkInterval)
+            deployment.undeployableInterval?.overlaps(checkInterval)
+        }
+
+        if (conflictingDeployments.isEmpty()) {
+            return true
+        }
+
+        [
+            'receiverDeployment.initialisationDateTime.conflictingDeployment',
+            deployment.receiver,
+            conflictingDeployments
+        ]
+    }
+
+    // Don't want to override 'equals()' as this causes unexpected behaviour with GORM.
+    boolean same(Object other) {
+        if (other == null) {
+            return false
+        }
+
+        return (this.receiver.same(other.receiver) && this.undeployableInterval == other.undeployableInterval)
+    }
+
+    String toString() {
+        return String.valueOf(receiver) + " - " + String.valueOf(undeployableInterval)
     }
 
     /**
      * Non-authenticated users can only see scrambled locations.
      */
-    Point getScrambledLocation()
-    {
+    Point getScrambledLocation() {
         return GeometryUtils.scrambleLocation(location ?: station?.location)
     }
 
-    double getLatitude()
-    {
+    double getLatitude() {
         return getScrambledLocation()?.coordinate?.y
     }
 
-    double getLongitude()
-    {
+    double getLongitude() {
         return getScrambledLocation()?.coordinate?.x
     }
 
-    private DateTime now()
-    {
+    private DateTime now() {
         return new DateTime()
     }
 
-    boolean isActive()
-    {
+    boolean isActive() {
         isActive(now())
     }
 
-    boolean isActive(dateTime)
-    {
-        if (!recovery)
-        {
+    boolean isActive(dateTime) {
+        if (!recovery) {
             return !deploymentDateTime.isAfter(dateTime)
         }
 
         return (!deploymentDateTime.isAfter(dateTime)) && dateTime.isBefore(recovery?.recoveryDateTime)
     }
+
+    def getUndeployableInterval() {
+        def startDateTime = initialisationDateTime ? initialisationDateTime : deploymentDateTime
+
+        if (startDateTime) {
+            if (recovery?.status == DeviceStatus.RECOVERED) {
+                return new Interval(startDateTime, recovery.recoveryDateTime)
+            }
+
+            if (recovery?.status) {
+                return new OpenInterval(startDateTime)
+            }
+        }
+    }
+
+    def getDeployedInterval() {
+        if (undeployableInterval instanceof Interval) {
+            return undeployableInterval
+        }
```

Comment:
I think `getDeloyedInterval()` can go actually - it doesn't seem to be used anywhere.  The important one is `undeployableInterval`, i.e. the period of time that this deployment causes the receiver to be undeployable (because it's actually in the water, i.e. still deployed, or lost, stolen, damaged etc).